examples/server/server.cpp

```diff
@@ -3602,15 +3625,31 @@ int main(int argc, char ** argv) {
         }
     };
 
-    const auto handle_props = [&ctx_server, &res_ok](const httplib::Request &, httplib::Response & res) {
+    std::mutex chat_templates_mutex;
+    std::optional<llama_chat_templates> chat_templates;
+
+    auto get_chat_templates = [&ctx_server, &chat_templates_mutex, &chat_templates]() -> const llama_chat_templates & {
```

Comment:
I think this function should be a member of `server_context`. Also, why do we have a lock `chat_templates_mutex` here? AFAIU `llama_chat_templates_from_model` must be thread-safe.

All params of `llama_chat_templates_from_model` are already `const` so it should be thread-safe. Otherwise, we're having a dangerous pointer casting somewhere that case away the `const`

Reply:
The mutex was to prevent concurrent initializations when triggered from request handlers, but now ditched / done synchronously in server_context::load_model, thanks!